src/libraries/System.IO.Packaging/src/System/IO/Packaging/ContentType.cs

```diff
@@ -490,34 +476,18 @@ private static void ValidateQuotedText(string quotedText)
         /// </summary>
         /// <param name="character">input character</param>
         /// <returns></returns>
-        private static bool IsAllowedCharacter(char character)
-        {
-            return Array.IndexOf(s_allowedCharacters, character) >= 0;
-        }
+        private static bool IsAllowedCharacter(char character) =>
+            Array.IndexOf(s_allowedCharacters, character) >= 0;
 
         /// <summary>
         /// Returns true if the input character is an ASCII digit or letter
         /// Returns false if the input character is not an ASCII digit or letter
         /// </summary>
         /// <param name="character">input character</param>
         /// <returns></returns>
-        private static bool IsAsciiLetterOrDigit(char character)
-        {
-            return (IsAsciiLetter(character) || (character >= '0' && character <= '9'));
-        }
-
-        /// <summary>
-        /// Returns true if the input character is an ASCII letter
-        /// Returns false if the input character is not an ASCII letter
-        /// </summary>
-        /// <param name="character">input character</param>
-        /// <returns></returns>
-        private static bool IsAsciiLetter(char character)
-        {
-            return
-                (character >= 'a' && character <= 'z') ||
-                (character >= 'A' && character <= 'Z');
-        }
+        private static bool IsAsciiLetterOrDigit(char character) =>
+            ((((uint)character - 'A') & ~0x20) < 26) ||
```

Comment:
This is no where near as clear as the previous. Not saying this isn't okay, but it might make sense to comment on the logic here. Is this a common way of computing this or is this a Toub original?

Reply:
It's not original, e.g. https://github.com/dotnet/runtime/blob/05cb7f5a6eefbcd0fbab3bb5924c07b188d12501/src/libraries/System.Private.Uri/src/System/UriHelper.cs#L574-L576.  We have many different IsAsciiLetterOrDigit methods floating around.

Reply:
> We have many different IsAsciiLetterOrDigit methods floating around.

Perhaps we should have one (with comment and tests) and reuse it everywhere?

Reply:
I agree we should have one. I think it should be be part of https://github.com/dotnet/runtime/issues/28230, which I'd like to see us prioritize for 7.